### DISCUSSION TITLE
ButtonWidget has protected access in [...]

### DISCUSSION THREAD

**Discussion by Dannyx1604** [FabricMC/fabric-api#3062] [Mod Dev Support] [answered]
I am currently trying to create a custom screen to edit my mod's settings in-game but when I try to add a ButtonWidget, IntelliJ gives me this error:

> 'ButtonWidget(int, int, int, int, net.minecraft.text.Text, net.minecraft.client.gui.widget.ButtonWidget.PressAction, net.minecraft.client.gui.widget.ButtonWidget.NarrationSupplier)' has protected access in 'net.minecraft.client.gui.widget.ButtonWidget'

This is my code(I _reused_ it from another mod, that other mod works):
`addDrawableChild(new ButtonWidget(this.width / 2 - 102, this.height / 6 + 24, 204, 20, Text.of("Setting 1: " + (setting1 ? "\u00a7aEnabled" : "\u00a7cDisabled")), button -> {
            DannyxMC.LOGGER.info("Toggling Setting 1");
            setting1 = !setting1;
            clearAndInit();
        },
                (NarrationSupplier) -> (MutableText) Text.of("Toggle Setting 1")
        ));`

Also my mod is in 1.19.4, the mod I used the code from is for 1.19.2
Edit: The only things I changed were the numbers for length and width + the variable names and the text displayed on the button

**Comment by Juuxel** [comment, marked as answer]
The relevant code changed between 1.19.2 and 1.19.4. You use a builder now:
```java
ButtonWidget.builder(
    Text.of("Setting 1: " + (setting1 ? "\u00a7aEnabled" : "\u00a7cDisabled")),
    button -> { DannyxMC.LOGGER.info("Toggling Setting 1"); setting1 = !setting1; clearAndInit(); }
)
    .dimensions(this.width / 2 - 102, this.height / 6 + 24, 204, 20)
    .narrationSupplier((NarrationSupplier) -> (MutableText) Text.of("Toggle Setting 1"))
    .build()
```

**Reply by Dannyx1604** [reply]
Ah, okay, thanks
